### DISCUSSION TITLE
Allow other jwpub files as public talk file

### DISCUSSION THREAD

**Discussion by e-Feitosa** [sircharlo/meeting-media-manager#3513] [Ideas] [answered]
### Is your idea related to a problem?

It's not a problem, but rather an idea or suggestion. See the screenshot below, I added the S-34 (without the MP), which is the list of public talk themes. This is not the S-34mp file that is in the other media section. This file with the list of themes includes all the updated talks until August 2022, unlike the other one which is updated until 2021.

![image](https://github.com/user-attachments/assets/41bc3698-1acb-4aa8-bb86-5b0d6eeb91ce)

### Describe your idea

Here it suggests inserting (adding / including) the S-34mp, this file was updated in 2021. I don't know if it would be interesting in the description to put to insert S-34 or put S-34 or S34mp?

![image](https://github.com/user-attachments/assets/3dc7fad0-6859-4686-b840-bce561867ad9)

### Describe alternatives you've considered

Another suggestion would be to include a note informing that both have the media and for the person to check the most updated S-34.

![image](https://github.com/user-attachments/assets/3df3b404-fed6-43a1-baac-ec8467b5f31e)

### Confirmations

- [X] I confirm that I have read the documentation website.
- [X] I confirm that I have searched through the existing issues to make sure my idea is not a duplicate.
- [X] I confirm that I have searched through the existing discussions to make sure my idea is not a duplicate.

**Comment by mtdvlpr** [comment]
A talk in S-34 can get updated without adding media to it. Does S-34 actually contain more media than S-34mp?

**Reply by e-Feitosa** [reply]
The S-34mp has 24 files with media, while the S-34 has approximately 64 files with media.

**Reply by sircharlo** [reply]
Are you referring to individual talk outlines? If so, they can be downloaded and added to the app as any other JWPUB file.

I'm not sure what the feature request is for 😅

**Reply by sircharlo** [reply]
I don't know what you mean by "the S-34".

**Reply by mtdvlpr** [reply]
S-34 contains all talk outlines and their media.

**Reply by sircharlo** [reply]
I don't think we have that in English. We have the S-99a (Public Talk Titles​—Listed by Subject).

**Comment by mtdvlpr** [comment]
I think we should keep the name as is, because the S-34mp file is specifically intended for sharing with the audio/video brothers, so they can get the standard media from public talks. The S-34 file should not be shared with those who are not authorized for it.

If anything, this is a bug on the branch side, because they're responsible for keeping the S-34mp file up-to-date.

**Comment by sircharlo** [comment]
If I understand correctly, the import works with either the S-34 or the S-34mp? If so, I'd leave everything as is. Like @mtdvlpr said, the S-34mp is really the one that's meant to be used by AV operators.

**Reply by e-Feitosa** [reply]
Yes, it was working normally for both files. I tested it in version 24.11.5, only the S-34 showed an error. The file is imported, but when adding an average of a certain sketch, it shows a missing media message. This has been happening in a new installation, and I also tested it a little while ago on a virtual machine. Even recognizing the file and showing the themes, it no longer downloads the media as before. I don't know what happened. On my PC, which has had this version since I opened this topic, it continues to work for any file (S-34 or S-34mp).

**Reply by mtdvlpr** [reply]
Which public talk are you trying to import?

**Reply by e-Feitosa** [reply]
Tested with several themes, yesterday a brother asked me for help with M3, I went to show this feature to him, and he downloaded the S-34 and added, but no theme was downloading media, is the message that is missing media, tested several, 19, 24, 70 etc. And when you add the S-34mp, works normal.

**Reply by sircharlo** [reply]
In that case, even more reason to leave the S-34mp label. 😄

**Reply by mtdvlpr** [reply]
I'll take a look at it tomorrow.

**Reply by mtdvlpr** [reply, marked as answer]
It's been fixed in https://github.com/sircharlo/meeting-media-manager/commit/189cadb6fc1eef76d114e95912146de2ef3e7aa9.

@sircharlo, I've changed the logic so that the cached public talk jwpub is stored under `S-34mp_<congId>`. This allows each congregation to save a different jwpub file and not interfere with the real `S-34mp_<lang>_0` folders. This way you can store any jwpub file as the public talk file.

If we don't want to support other files like `S-34`, we should get the publication from the database and verify that it's really the `S-34mp` file of the correct language before accepting it. I don't think it hurts to allow the user to set a different jwpub as the default public talk file.
